MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/location/LocationLayerController.java

```diff
@@ -276,6 +301,30 @@ private void updateAccuracyRadius(float accuracy) {
     refreshSource();
   }
 
+  /**
+   * Updates the {@link LocationComponentConstants#PROPERTY_PULSING_RADIUS} property value and refreshes
+   * the LocationComponent source. This leads to a smooth update to the visuals of the pulsing
+   * LocationComponent circle.
+   *
+   * @param radius The new radius in the animation.
+   */
+  private void updatePulsingLocationCircleRadius(float radius) {
+    locationFeature.addNumberProperty(PROPERTY_PULSING_RADIUS, radius);
+    refreshSource();
+  }
+
+  /**
+   * Updates the {@link LocationComponentConstants#PROPERTY_PULSING_OPACITY} property value and refreshes
+   * the LocationComponent source. This leads to a smooth update to the visuals of the pulsing
+   * LocationComponent circle. This is used if the fade option is set to true while setting pulsing options.
+   *
+   * @param opacity The new opacity in the animation.
+   */
+  private void updatePulsingLocationCircleOpacity(float opacity) {
+    locationFeature.addNumberProperty(PROPERTY_PULSING_OPACITY, opacity);
+    refreshSource();
+  }
```

Comment:
Both `updatePulsingLocationCircleRadius` and `updatePulsingLocationCircleOpacity` are invoked (conditionally) from `pulsingCircleRadiusListener`, and only from there. So, instead of both of them being separate, and invoking `refreshSource` twice, we could make them a single method that based on the conditions changes the radius property, opacity property, and only then updates the source.

Reply:
Got it. Thanks for the clarification. Just pushed a commit that consolidates this.